src/main/java/net/neoforged/neoform/runtime/actions/ExtensibleClasspath.java

```diff
@@ -33,12 +33,25 @@ public List<ClasspathItem> getEffectiveClasspath() {
 
     public void addMinecraftLibraries(Collection<MinecraftLibrary> libraries) {
         for (var library : libraries) {
-            if (library.rulesMatch() && library.getArtifactDownload() != null) {
+            if (library.rulesMatch() && library.getArtifactDownload() != null && !containsLibrary(library)) {
```

Comment:
It might be good to log that the version is being overridden (at an info level).